packages/interactivity/src/init.js

```diff
@@ -30,17 +30,47 @@ function yieldToMain() {
 
 // Initialize the router with the initial DOM.
 export const init = async () => {
+	const pendingNodes = new Set();
+
+	const intersectionObserver = new window.IntersectionObserver(
+		async ( entries ) => {
+			for ( const entry of entries ) {
+				if ( ! entry.isIntersecting ) {
+					continue;
+				}
+
+				const node = entry.target;
+				intersectionObserver.unobserve( node );
+				pendingNodes.delete( node );
+				if ( pendingNodes.size === 0 ) {
+					intersectionObserver.disconnect();
+				}
```

Comment:
This would be problematic for islands that are added dynamically during client-side navigations. The IntersectionObserver may need to remain persistently, ir `init()` isn't called again after new islands are added.